### ISSUE TITLE
add daemon or watch mode equivalent of `tye run`

### ISSUE THREAD

**Issue by rynowak** [dotnet/tye#13]
The idea is that we can leave the host and dashboard running, and services can drop in and out.

**Comment by ChadJessup** [comment]
Just a thought, but could the dashboard be in a container that is automatically added to a tye.yaml?

It'd be useful to have a preexisting and non-trivial working component in a new tye project by default. Especially one that uses some of the newer goodies like Blazor, and potentially uses the very features that tye offers for microservices...

**Comment by davidfowl** [comment]
What does the dashboard being in a container gain you?

**Comment by ChadJessup** [comment]
> What does the dashboard being in a container gain you?

No must-haves, but aside from the built-in sample I mentioned above, I think it'd also be useful to a user of tye for a few other reasons:

* Bootstrap: Bellwether service showing that everything is working (docker/dotnet core/kubernetes/tye/permissions/registry)
* Bootstrap: A built-in dashboard is a great bootstrapping tool.
* Decouples tye CLI from a web page - this has confused me a few times, as closing a terminal window rarely impacts a web page in my experience.
* Can restart/shutdown the dashboard directly if it's not needed.
* Can deploy the dashboard to k8s as well with `deploy`, which could keep the same experience as local dev, and give immediate access to logs, health (#19), metrics after deployment. Potentially even use k8s's downward APIs to help manage things if something like replication, and lifecycle control is planned from the dashboard itself (shutdown, restart, spin up five of these, etc). 
* I imagine tye CLI could be used to point to different hosts like kubectl's --context flag. But, I'm not sure how intertwined the dashboard and hosting functionality are.

Cool thing is tye already knows how to create containers and helm charts, could be like an embedded resource thing that turns into a container...but, just thinking out loud on that 🤔.

**Comment by davidfowl** [comment]
> Decouples tye CLI from a web page - this has confused me a few times, as closing a terminal window rarely impacts a web page in my experience.

Yes I think that's one of the reasons for making that kind of experience in the first place. We'll have to do something like this anyways to make it run in the cluster.

I'm hesitant to rely on docker for core functionality but it might be fine as a mode.

**Comment by staff0rd** [comment]
Presume this would watch `tye.yaml` for changes and recreate containers based on changes to the .yml, rather than currently having to kill/restart `tye run`

**Comment by rynowak** [comment]
Yes, we super want this 👍 

The work that I did this milestone with diagnostics integration makes it even more important because those things are slow to start.
